instrumentation/java-http-server/library/src/main/java/io/opentelemetry/instrumentation/httpserver/OpenTelemetryService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.httpserver;
+
+import com.sun.net.httpserver.Filter;
+import com.sun.net.httpserver.HttpExchange;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import java.io.IOException;
+
+/** Decorates an {@link HttpServer} to trace inbound {@link HttpExchange}s. */
+final class OpenTelemetryService extends Filter {
+
+  private final Instrumenter<HttpExchange, HttpExchange> instrumenter;
+
+  OpenTelemetryService(Instrumenter<HttpExchange, HttpExchange> instrumenter) {
+
+    this.instrumenter = instrumenter;
+  }
+
+  @Override
+  public void doFilter(HttpExchange exchange, Chain chain) throws IOException {
+
+    Context parentContext = Context.current();
+    if (!instrumenter.shouldStart(parentContext, exchange)) {
+      chain.doFilter(exchange);
+      return;
+    }
+
+    Context context = instrumenter.start(parentContext, exchange);
+
+    try (Scope ignored = context.makeCurrent()) {
+      chain.doFilter(exchange);
+    } finally {
+      instrumenter.end(context, exchange, exchange, null);
+    }
```

Comment:
usually we also try to capture the exceptions from `doFilter` call with something like
```java
    Throwable error = null;
    try (Scope ignored = context.makeCurrent()) {
      chain.doFilter(exchange);
    } catch (Throwable t) {
      error = t;
      throw t;
    } finally {
      if (httpRouteSupport.hasMappings()) {
        HttpServerRoute.update(context, CONTROLLER, httpRouteSupport::getHttpRoute, request);
      }
      instrumenter.end(context, exchange, exchange, error);
    }
```

Reply:
Throwing an exception from a handler/filter causes a socket hangup